File: Desafios Aleatorios/Java/exe01.java
import java.util.Scanner;

public class exe01 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        double salario;
        int tempo;
        float desempenho;
        int bonusTempo;
        int bonusDesempenho;

        salario = ObterSalario(scanner);
        if (!ValidarSalario(salario)){
            System.exit(1);
        }

        tempo = ObterTempoServico(scanner);
        if (!ValidarTempoServico(tempo)){
            System.exit(1);
        }

        desempenho = ObterNotaDesempenho(scanner);
        if (!ValidarNotaDesempenho(desempenho)){
            System.exit(1);
        }

        bonusTempo = CalcularBonusTempo(tempo);
        bonusDesempenho = CalcularBonusDesempenho(bonusTempo, desempenho);

        ExibirResultado(salario, bonusDesempenho);

        scanner.close();
    }

    public static void ExibirResultado(double salario, int bonusDesempenho){
        if (bonusDesempenho == 0){
            System.out.println("Sem direito a bônus");
        } else {
            double bonusFinal = salario * (bonusDesempenho / 100.0);
            System.out.println("Bônus Final: " + bonusFinal);
        }
    }

    public static int CalcularBonusDesempenho(int bonusTempo, float notaDesempenho){
        if (notaDesempenho < 4){
            return 0;
        } else if (notaDesempenho >= 4 && notaDesempenho <= 7){
            return bonusTempo;
        } else {
            return bonusTempo * 2;
        }
    }

    public static int CalcularBonusTempo(int tempo){
        if (tempo < 3){
            return 0;
        } else if (tempo >= 3 && tempo <= 5){
            return 5;
        } else {
            return 10;
        }
    }

    public static double ObterSalario(Scanner scanner) {
        System.out.print("Informe o salario Base do funcionario: ");
        return scanner.nextDouble();
    }

    public static boolean ValidarSalario(double salario){
        if (salario <= 0){
            System.out.println("ERRO: Salário não pode ser negativo.");
            return false;
        }
        return true;
    }

    public static int ObterTempoServico(Scanner scanner){
        System.out.print("Informe o tempo de serviço: ");
        return scanner.nextInt();
    }

    public static boolean ValidarTempoServico(int tempo){
        if (tempo < 0){
            System.out.println("ERRO: Tempo de serviço não pode ser negativo.");
            return false;
        }
        return true;
    }

    public static float ObterNotaDesempenho(Scanner scanner){
        System.out.print("Informe a nota de desempenho: ");
        return scanner.nextFloat();
    }

    public static boolean ValidarNotaDesempenho(float nota){
        if (nota < 0 || nota > 10){
            System.out.println("ERRO: Nota de desempenho deve estar entre 0 e 10.");
            return false;
        }
        return true;
    }
}

File: Desafios Aleatorios/Python/exe01.py
def obter_salario() -> float:
    salario = float(input("Informe o salario: "))
    if salario <= 0:
        print("Erro: Dados Invalidos, salario deve ser maior que 0.")
        exit()
    return salario

def obter_tempo_servico() -> int:
    tempo = int(input("Informe o tempo de serviço: "))
    if tempo < 0:
        print("Erro: Dados Invalidos, Tempo de Serviço não pode ser negativo.")
        exit()
    return tempo

def obter_nota_desempenho() -> float:
    # considerando que essa nota pode ser um valor decimal
    # Exemplo: 9.5
    nota = float(input("Informe a nota de desempenho: "))
    if nota < 0 or nota > 10:
        print("Erro: Dados invalidos, Nota de desempenho deve estar entre 0 e 10.")
        exit()
    return nota

def calcular_bonus_tempo(tempo: int) -> int:
    if tempo < 3:
        return 0
    elif tempo >= 3 and tempo <= 5:
        return 5
    else:
        return 10

def calcular_bonus_nota(nota: float, bonus_tempo: int) -> int:
    if nota < 4:
        return 0
    elif nota >= 4 and nota <= 7:
        return bonus_tempo
    else:
        return bonus_tempo * 2

def exibir_resultado(salario: float, bonus: int) -> None:
    if bonus == 0:
        print("Sem direito a bônus.")
        return
    bonus_final = salario * (bonus / 100)
    print(f"Bônus final: {bonus_final:,.2f}")

if __name__ == "__main__":
    # entrada e validação das entradas
    salario = obter_salario()
    tempo = obter_tempo_servico()
    nota = obter_nota_desempenho()

    # Processamento
    bonus_tempo = calcular_bonus_tempo(tempo)
    bonus_nota = calcular_bonus_nota(nota, bonus_tempo)

    # saida 
    exibir_resultado(salario, bonus_nota)